detailed_spec/requirements.tex

```diff
@@ -1,30 +1,38 @@
 \chapter{Requirements}
 
-\todo{Tohle by mozna nebylo spatny rovnou pojmenovat nejak jako `Features', `API' nebo mozna `Interfaces'.}
+This chapters outlines the main features that the project should implement to be a working product. All the features below have been discussed with professional HLASM developers and have been agreed on by the company.
 
--co ten nas produkt ma byt
-vseobecne zhrnutie
+The final product is a VS Code extension, downloadable from the Market Place. The extension contains all executables/libraries that are needed for the project to work correctly on the most popular platforms. No other prerequisites should be needed.
 
-...je to extension ... doda support pre ... 
--cela tato sekcia uz je popisana niekde na CA wiki, mozno dobry zaklad
+Modularity of the project is another important requirement. 
+
+The parsing library provides 2 kinds of API: complex one that mimicks the LSP specification and a simple one that accepts text and dependency-resolver object and returns diagnostics. 
+
+The language server implements the LSP standard, hence it can be easily reintegrated within other IDEs such as Eclipse Che.
 
 \section{Language features}
--zoznam veci jazyka co podporujeme
+This part provides a brief overview of the parts of HLASM that will be implemented in  the project.
+
+First of all, the project recognizes the syntax of HLASM (including continuations) and parses it into predefined structures. 
+
+It is able to interpret high-level parts of the assembler such as the conditional assembly instructions for the code generation, assembler instruction for the code layout determination and the macro expansion.
 
+For the dependency search, a configuration file is necessary to provide locations of depending files. This configuration procedure is similar to the one used on real mainframes.
```

Comment:
Mozno sem hodit presnu specifikaciu tych configov? Tu uz mam napisanu na CA wiki urcite, zajtra sa k nej dostaneme

Reply:
spomenul by som aj to predkonfigurovanie tych configov na ktorom teraz robis v praci

Reply:
jj, to mozme hodit do code snippetu a trochu to vysvetlit.